File: [23.12]Design-Pattern/Java/src/staticFactoryMethod/Drink.java
package staticFactoryMethod;

public class Drink {
    private Drink(){}

    public static Drink from(String msg){
        System.out.println("make Drink" + msg);
        return new Drink();
    }

    public static Drink of(String... msg){
        System.out.println("make Drink");
        for (String str :
                msg) {
            System.out.println(str);
        }
        return new Drink();
    }

    public static Drink getInstance(){
        return new Drink();
    }

    public static Drink newInstance(){
        return new Drink();
    }

    public static String getString(){
        return "Drink";
    }

    public static String newString(){
        return "Drink";
    }

}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/creational/staticFactoryMethod/Drink.kt
package creational.staticFactoryMethod

class Drink {
    companion object{
        fun from():Drink{
            return Drink()
        }

        fun of(): Drink{
            return Drink()
        }

        fun valueOf():Drink{
            return Drink()
        }

        fun getInstance(): Drink{
            return Drink()
        }

        fun newInstance(): Drink{
            return Drink()
        }

        fun getString():String{
            return "Drink"
        }

        fun newString():String{
            return "Drink"
        }




    }
}